landlab/components/river_bed_dynamics/river_bed_dynamics.py

```diff
@@ -0,0 +1,3433 @@
+            the current surface GSD, and the bed load GSD and rate.
+        track_stratigraphy : bool, optional
+            If ``True``, the component stores the GSD of each layer at every node.
+            This is computationally demanding as it needs to read/write data
+            at every time step. Recommended only when cycles of erosion and
+            deposition are frequent and very important.
+        number_cycles_to_process_stratigraphy : int, optional
+            If ``track_stratigraphy`` is ``True``, data will be read and stored every
+            ``'number_cycles_to_process_stratigraphy'`` time steps. Must be larger
+            or equal to 1. All data are written to the hard drive. It does not use
+            *Landlab* layers.
+        dt: float, optional
+            Time step in seconds. When this component is coupled to a flow model,
+            it is dynamically updated.
+        alpha : float, optional
+            An upwinding coefficient for a central difference scheme when
+            updating the bed GSD - default value is 1.0 - a value of 0.5
+            generates a central differences scheme.
+        bed_surface__elevation_fixed_node: int, optional
+            Sets a node as a fixed elevation.
+            Units: - , mapping: node
+        sediment_transport__bedload_rate_imposed_link: float, optional
+            Sets the sediment transport rate per unit width where sediment supply is imposed.
+            Units: m^2/s, mapping: link
+        bed_surface__grain_size_distribution_fixed_node: int, optional
+            Sets the locations (nodes) that do not change its GSD in time
+            Units: - , mapping: node
+        bed_surface__grain_size_distribution_location_node:  int, optional
+            Sets the location at each node in which the GSD applies
+            Units: - , mapping: node
+        sediment_transport__bedload_gsd_imposed_link: float, optional
+            Sets the sediment transport GSD where sediment supply is imposed
+            Units: - , mapping: link
+        surface_water__velocity_previous_time_link: float, optional
+            Speed of water flow above the surface in the previous time step
+            Units: m/s, mapping: link
+        current_t : float, optional
+            Current simulation time or elapsed time. It does not update automatically
+            Units: s
+        """
+        super().__init__(grid)
+
+        self._g = scipy.constants.g  # Acceleration due to gravity (m/s**2).
+        self._rho = rho
+        self._rho_s = rho_s
+        self._R = (rho_s - rho) / rho
+        self._shear_stress_star_rsgo = 0.0386  # Reference dimensionless shear
+        # stress for the median size
+        self._beta = 0.0951  # Coefficient for the hiding function
+        if gsd is None:
+            gsd = np.array([[32, 100], [16, 25], [8, 0]])
+        self._gsd = gsd
+        self._bedload_equation = bedload_equation
```

Comment:
Instead of setting ``self._bedload_equation`` to the name of the equation, how about setting it to the function?
```python
if bedload_equation == "Parker1990":
    self._bedload_equation = self.bedload_equation_Parker_1990
elif bedload_equation == "MPM":
    self._bedload_equation = self.bedload_equation_MeyerPeter_Muller
elif bedload_equation == "FLvB":
    self._bedload_equation = self.bedload_equation_FernandezLuque_VanBeek
elif bedload_equation == "WilcockAndCrowe":
    self._bedload_equation = self.bedload_equation_Wilcock_Crowe_2003
```
Then when you want to call the equation further down, you can just use ``self._bedload_equation()``.

You could save the name of the equation as a separate variable, if you need (as it looks like you do later on).

Reply:
Thanks, done!